opencl/opencl-aot/source/main.cpp

```diff
@@ -349,6 +356,9 @@ int main(int Argc, char *Argv[]) {
   cl::list<std::string> OptBuildOptions("bo", cl::ZeroOrMore,
                                         cl::desc("Set OpenCL build options"),
                                         cl::value_desc("build options"));
+  cl::opt<bool, true> OptVerbose("verbose", cl::desc("Show verbose logs"),
+                                 cl::location(verbose));
+  cl::alias OptV("v", cl::aliasopt(OptVerbose));
```

Comment:
Since verbose output is now an optional behavior, how can it be turned off? If it cannot be turned off, then what is the reason for adding the option, compatibility with some other tool?

Reply:
The verbose log is NOT optional behavior now. This patch is to turn off it by default and provide an option to turn on.

Reply:
> The verbose log is NOT optional behavior now. 

Ok, got it. It think I got confused by `true` template parameter to `opt`, could you please clarify what it is for?

Reply:
https://llvm.org/docs/CommandLine.html#internal-vs-external-storage
> By default, all command line options automatically hold the value that they parse from the command line.

> ```cpp
> bool DebugFlag;                  // the actual value
> static cl::opt<bool, true>       // The parser
> Debug("debug", cl::desc("Enable debug output"), cl::Hidden, cl::location(DebugFlag));
> ```
> we specify “`true`” as the second argument to the `cl::opt` template, indicating that the template should not maintain a copy of the value itself. In addition to this, we specify the `cl::location` attribute, so that `DebugFlag` is automatically set.

Here we want to store the option value in global variable `verbose` by specifying `cl::location(verbose)` attribute, which requires the second argument to the `cl::opt` template to be `true`.

Reply:
If we have versioning in the future, option `-v` can be accidentally used in the wrong way by users because it is expected that `-v` is alias for `--version`. Do we need shorter alias for `--verbose`?

Reply:
We can align the `-v` behavior with clang whose help doc says it shows verbose output. In fact, clang always emits version information with `-v` option whether there is something to be done or not (per @mdtoguchi).

If we have versioning in the future, we can prepend the version information to the verbose output. For now, I prefer use `-v` as an alias for ``--verbose` just like what I'm doing.

Reply:
@AlexeySachkov Could you please confirm that this comment is resolved?